services/src/datasets/external/netcdfcf/ebvportal_provider.rs

```diff
@@ -611,30 +535,138 @@ impl LayerCollectionProvider for EbvPortalDataProvider {
     async fn load_layer(&self, id: &LayerId) -> Result<Layer> {
         let ebv_id: EbvCollectionId = EbvCollectionId::from_str(&id.0)?;
 
-        match &ebv_id {
-            EbvCollectionId::Entity {
-                class: _,
-                ebv: _,
-                dataset,
-                groups,
-                entity,
-            } => {
-                let ebv_hierarchy = self
-                    .ebv_api
-                    .get_ebv_subdatasets(
-                        NetCdfCfDataProviderPaths {
-                            provider_path: self.netcdf_cf_provider.path.clone(),
-                            overview_path: self.netcdf_cf_provider.overviews.clone(),
-                        },
-                        dataset,
-                    )
-                    .await?;
+        let EbvCollectionId::Entity {
+            class,
+            ebv,
+            dataset,
+            groups,
+            entity,
+        } = ebv_id
+        else {
+            return Err(Error::InvalidLayerId);
+        };
+
+        let dataset_path = self
+            .ebv_api
+            .get_dataset_metadata(&dataset)
+            .await
+            .map(|dataset| PathBuf::from(dataset.dataset_path.trim_start_matches('/')))?;
+
+        let layer_id = netcdf_entity_to_layer_id(&dataset_path, &groups, entity);
+
+        let mut layer = self.netcdf_cf_provider.load_layer(&layer_id).await?;
 
-                layer_from_netcdf_overview(EBV_PROVIDER_ID, id, ebv_hierarchy.tree, groups, *entity)
+        replace_id_in_layer(&mut layer.id, &class, &ebv, &dataset)?;
+
+        Ok(layer)
+    }
+}
+
+/// We use the [`NetCdfCfDataProvider`] to answer requests for metadata.
+/// As this returns its [`DataProviderId`] as well as [`NetCdfLayerCollectionId`]s.
+/// Thus, we make subtle modifications to return [`EBV_PROVIDER_ID`] as well as [`EbvCollectionId`]s.
+fn replace_ids_in_collection_items(
```

Comment:
is it possible to just set the correct ids in the first place instead of replacing them? Like pass it as a a parameter ind get_dataset_metadata?

Reply:
[83ce94d](https://github.com/geo-engine/geoengine/pull/931/commits/83ce94d0db8520ffe15396aa513da000a4b30d39)